lib/ansible/modules/packaging/os/yum.py

```diff
@@ -757,27 +756,25 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos, i
                 # Fail on invalid urls:
                 if '://' in spec and ('No package %s available.' % spec in out or 'Cannot open: %s. Skipping.' % spec in err):
                     module.fail_json(msg='Package at %s could not be installed' % spec, rc=1, changed=False)
-        if (rc != 0 and 'Nothing to do' in err) or 'Nothing to do' in out:
-            # avoid failing in the 'Nothing To Do' case
-            # this may happen with an URL spec.
-            # for an already installed group,
-            # we get rc = 0 and 'Nothing to do' in out, not in err.
-            rc = 0
-            err = ''
-            out = '%s: Nothing to do' % spec
-            changed = False
 
         res['rc'] = rc
         res['results'].append(out)
         res['msg'] += err
+        res['changed'] = True
```

Comment:
It might be good to use `res['stderr']` everywhere in the module. Are you ok with me addressing that in a separate PR?